src/storage.ts

```diff
@@ -0,0 +1,60 @@
+interface PersistantStorage {
+  [x: string]: any; // this makes sure that the storage can be accessed with any key
+
+  /**
+   * @param key - The key to store the data under
+   * @param data - The data to store
+   * @throws {Error} If the data could not be stored
+   */
+  store(key: string, value: any): void;
+
+  /**
+   * @param key - The key to retrieve the data from
+   * @returns The data stored under the key
+   * @throws {Error} If the data could not be retrieved
+   */
+  retrieve(key: string): any;
+}
```

Comment:
this also should define remove(key: string) and maybe clear()

Reply:
clear() implmented in logout pull request